tools/walletextension/multi_acc_helper.go

```diff
@@ -142,30 +142,39 @@ func parseParams(args []interface{}) (map[string]interface{}, error) {
 }
 
 // proxyRequest will try to identify the correct EncRPCClient to proxy the request to the Obscuro node, or it will attempt
-//
 //	the request with all clients until it succeeds
-func proxyRequest(rpcReq *rpcRequest, rpcResp *interface{}, accClients map[common.Address]*rpcclientlib.EncRPCClient) error {
+func proxyRequest(rpcReq *rpcRequest, rpcResp *interface{}, we *WalletExtension) error {
 	// for obscuro RPC requests it is important we know the sender account for the viewing key encryption/decryption
-	suggestedClient := suggestAccountClient(rpcReq, accClients)
+	suggestedClient := suggestAccountClient(rpcReq, we.accountClients)
 
 	var err error
-	if suggestedClient != nil {
+	switch {
+	case suggestedClient != nil: // use the suggested client if there is one
 		// todo: if we have a suggested client, should we still loop through the other clients if it fails?
 		// 		The call data guessing won't often be wrong but there could be edge-cases there
-		err = executeCall(suggestedClient, rpcReq, rpcResp)
-	} else {
-		// we attempt the request with every client until we have a successful execution
-		log.Info("appropriate client not found, attempting request with up to %d clients", len(accClients))
-		for _, client := range accClients {
+		return executeCall(suggestedClient, rpcReq, rpcResp)
+
+	case len(we.accountClients) > 0: // try registered clients until there's a successful execution
+		log.Info("appropriate client not found, attempting request with up to %d clients", len(we.accountClients))
+		for _, client := range we.accountClients {
 			err = executeCall(client, rpcReq, rpcResp)
 			if err == nil || errors.Is(err, rpcclientlib.ErrNilResponse) {
 				// request didn't fail, we don't need to continue trying the other clients
-				break
+				return nil
 			}
 		}
-	}
+		// every attempt errored
+		return err
 
-	return err
+	default: // no clients registered, use a temporary one
+		// TODO: We might consider having a permanent, no-account client to avoid the overhead of connection setup
+		client, err := rpcclientlib.NewNetworkClient(we.hostAddr)
+		if err != nil {
+			return fmt.Errorf("unable to create temporary client for request - %w", err)
+		}
+
```

Comment:
I think it would be better to check whether this is going to fail before calling the host, like:

```
if isSensitive(method) {
    return Error("method %s cannot be called with an unauthorised client but no signed viewing keys found")
}
```

You'll need to make the `sensitiveMethods` list public in the `rpcclientlib/client.go` to make that isSensitive func.

Reply:
Yep, makes sense 👍